object/object.go

```diff
@@ -121,3 +131,31 @@ func SplitSlabs(slabs []Slab, lengths []int) [][]SlabSlice {
 func SingleSlabs(slabs []Slab, length int) []SlabSlice {
 	return SplitSlabs(slabs, []int{length})[0]
 }
+
+type rekeyStream struct {
+	key []byte
+	c   *chacha20.Cipher
+
+	counter uint64
+	nonce   uint64
+}
+
+func (rs *rekeyStream) XORKeyStream(dst, src []byte) {
+	rs.counter += uint64(len(src))
```

Comment:
if we only increment the counter after comparing we don't have to subtract `len(src)` further down?